packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -451,4 +451,22 @@ class _RenderFlexibleSpaceHeaderOpacity extends RenderOpacity {
 
   @override
   bool get isRepaintBoundary => false;
+
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child == null) {
+      return;
```

Comment:
Why do we not need to null out the layer in this case (i.e. if you switch from having a child drawn into a layer to not having one)?

Reply:
🤷 This follows the behavior of the old opacity layer.